public/src/managers/ServerProvider.ts

```diff
@@ -81,114 +81,116 @@ export class ServerProvider implements IUserProvider, ICourseProvider {
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return [];
         }
         return result.data.getEnrollmentsList();
     }
 
     public async getUsersForCourse(
         course: Course, noGroupMembers?: boolean,
-        state?: Enrollment.UserStatus[]): Promise<Enrollment[]> {
+        status?: Enrollment.UserStatus[]): Promise<Enrollment[]> {
 
-        const result = await this.grpcHelper.getEnrollmentsByCourse(course.getId(), noGroupMembers, state);
-        if (result.status.getCode() !== 0 || !result.data) {
+        const result = await this.grpcHelper.getEnrollmentsByCourse(course.getId(), noGroupMembers, status);
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return [];
         }
         return result.data.getEnrollmentsList();
     }
 
     public async getAssignments(courseID: number): Promise<Assignment[]> {
         const result = await this.grpcHelper.getAssignments(courseID);
-        if (result.status.getCode() !== 0 || !result.data) {
+        if (!this.responseCodeSuccess(result) || !result.data) {
             return [];
         }
         return result.data.getAssignmentsList();
     }
 
     public async addUserToCourse(course: Course, user: User): Promise<boolean> {
         const result = await this.grpcHelper.createEnrollment(course.getId(), user.getId());
-        return result.status.getCode() === 0;
+        return this.responseCodeSuccess(result);
     }
 
-    public async changeUserState(link: Enrollment, state: Enrollment.UserStatus): Promise<Status> {
-        link.setStatus(state);
-        const result = await this.grpcHelper.updateEnrollment(link);
+    public async changeUserStatus(enrollment: Enrollment, status: Enrollment.UserStatus): Promise<Status> {
+        const originalStatus = enrollment.getStatus();
+        enrollment.setStatus(status);
+        const result = await this.grpcHelper.updateEnrollment(enrollment);
+        if (!this.responseCodeSuccess(result)) {
+            enrollment.setStatus(originalStatus);
+        }
         return result.status;
     }
 
     public async approveAll(courseID: number): Promise<boolean> {
         const result = await this.grpcHelper.updateEnrollments(courseID);
-        if (result.status.getCode() !== 0 || !result.data) {
-            return false;
-        }
-        return true;
+        return result.data ? this.responseCodeSuccess(result) : false;
     }
 
     public async isAuthorizedTeacher(): Promise<boolean> {
         const result = await this.grpcHelper.isAuthorizedTeacher();
-        if (result.status.getCode() !== 0 || !result.data) {
-            return false;
+        if (this.responseCodeSuccess(result) && result.data) {
+            return result.data.getIsauthorized();
         }
-        return result.data.getIsauthorized();
+        return false;
+
```

Comment:
Ref to comment below with an updated method that also checks `result.data`, this pattern becomes:
```typescript
if (this.responseSuccess(result)) {
    return result.data.getIsauthorized();
}
return false;
```